src/main/java/org/opengis/cite/iso19142/basic/filter/spatial/BBOXTests.java

```diff
@@ -179,6 +179,21 @@ public void bboxWithDefaultExtent(ProtocolBinding binding, QName featureType) {
         }
     }
 
+    /**
+     * Work around for Extents{@link #envelopeAsGML(Envelope)} only using 2 decimals
+     */
+    public Document envelopeAsGML(Envelope extent) {
+        double[] low = new double[extent.getDimension()];
+        double[] high = new double[extent.getDimension()];
+        for (int i = 0; i < extent.getDimension(); i++) {
+            low[i] = extent.getMinimum(i) - ENVELOPE_EXPANSION;
```

Comment:
Thinking out loud, maybe we want to also check if the CRS is geographic, and if so, make sure the result is within the normal coordinate limits.